### ISSUE TITLE
deep scan never finishes: UnicodeDecodeError

### ISSUE THREAD

**Issue by hlef** [bleachbit/bleachbit#336]
Hi,

[#895979](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=895979) was recently reported on the Debian bug tracking system:

```
Package: bleachbit
Version: 2.0-2
Severity: normal

Bleachbit hangs while doing deep scan

Console output:

Traceback (most recent call last):
  File "/usr/share/bleachbit/bleachbit/Worker.py", line 254, in run
    for dummy in self.run_deep_scan():
  File "/usr/share/bleachbit/bleachbit/Worker.py", line 313, in run_deep_scan
    for path in ds.scan():
  File "/usr/share/bleachbit/bleachbit/DeepScan.py", line 85, in scan
    for (dirpath, dirnames, filenames) in normalized_walk(top):
  File "/usr/share/bleachbit/bleachbit/DeepScan.py", line 59, in normalized_walk
    for result in os.walk(top, **kwargs):
  File "/usr/lib/python2.7/os.py", line 296, in walk
    for x in walk(new_path, topdown, onerror, followlinks):
  File "/usr/lib/python2.7/os.py", line 296, in walk
    for x in walk(new_path, topdown, onerror, followlinks):
  File "/usr/lib/python2.7/os.py", line 296, in walk
    for x in walk(new_path, topdown, onerror, followlinks):
  File "/usr/lib/python2.7/os.py", line 296, in walk
    for x in walk(new_path, topdown, onerror, followlinks):
  File "/usr/lib/python2.7/os.py", line 296, in walk
    for x in walk(new_path, topdown, onerror, followlinks):
  File "/usr/lib/python2.7/os.py", line 286, in walk
    if isdir(join(top, name)):
  File "/usr/lib/python2.7/posixpath.py", line 73, in join
    path += '/' + b
UnicodeDecodeError: 'utf8' codec can't decode byte 0xed in position 18: invalid continuation byte
```

This may be someway related to #209, but since the stacktrace is different, I'm opening a new bug report.

Thanks for your work.

**Comment by az0** [comment]
@hlef The stacktrace is similar, so the fix may be the same. Either way, opening another bug ticket is fie.

@vasyugan I understand you are already using Unicode, but would you please try this fix?

Either 
1. Apply the patch from commit 7f7451f
2. Download the whole [DeepScan.py file](https://raw.githubusercontent.com/bleachbit/bleachbit/7f7451ffaf6a0a045509dd228dd30aab4b7f6065/bleachbit/DeepScan.py) and overwrite your copy. 

These two options above should be equivalent.

Please let me know how it goes. I

**Comment by az0** [comment]
The fix was published in [BleachBit 2.1 beta](https://www.bleachbit.org/news/bleachbit-21-beta). Soon the fix will be published also in BleachBit 2.2 final.
